src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/LaunchProfilesDebugLaunchProvider.cs

```diff
@@ -48,9 +51,22 @@ internal class LaunchProfilesDebugLaunchProvider : DebugLaunchProviderBase, IDep
         /// <summary>
         /// Called by CPS to determine whether we can launch
         /// </summary>
-        public override Task<bool> CanLaunchAsync(DebugLaunchOptions launchOptions)
+        public override async Task<bool> CanLaunchAsync(DebugLaunchOptions launchOptions)
         {
-            return TplExtensions.TrueTask;
+            try
+            {
+                // Launch providers to enforce requirements for debuggable projects
+                await QueryDebugTargetsInternalAsync(launchOptions, true);
+            }
+            catch(Exception e)
+            {
+                if (string.Compare(e.Message, VSResources.ProjectNotRunnableDirectly, StringComparison.OrdinalIgnoreCase) == 0)
```

Comment:
>Compare [](start = 27, length = 7)

Should we have a StringComparer (defined in the StringComparers class) for error messages?

Reply:
Which StringComparer property is the one that applies to e.Message ?